hydra-node/test/Hydra/Model.hs

```diff
@@ -474,11 +488,17 @@ seedWorld seedKeys seedCP = do
       pure (party, testNode, nodeThread)
     pure (res, tickThread)
 
-  modify $ \n ->
-    n
-      { nodes = Map.fromList $ (\(p, t, _) -> (p, t)) <$> fst nodes
-      , threads = snd nodes : ((\(_, _, t) -> t) <$> fst nodes)
-      }
+  s <- ask
+  lift $
+    atomically $
+      modifyTVar (nodesState s) $ \n ->
```

Comment:
Can we define a `modifyRunState :: MonadSTM m => (Nodes m -> Nodes m) -> m` instead? It would hide the fact that we are actually using a `ReaderT TVar` under the hood and be easier to update in case this changes further.

Reply:
or an `-> STM` variant where you would have control over the atomically section.